rpm-package/jellyfin.spec

```diff
@@ -0,0 +1,152 @@
+Source1:        jellyfin.service
+Source2:        jellyfin.env
+Source3:        jellyfin.sudoers
+Source4:        restart.sh
+Source5:        https://github.com/mono/taglib-sharp/archive/%{taglib_commit}/taglib-sharp-%{taglib_shortcommit}.tar.gz
+Source6:        update-db.sh
+Source7:        jellyfin-firewalld.xml
+
+%{?systemd_requires}
+BuildRequires:  systemd
+Requires(pre):  shadow-utils
+BuildRequires:  libcurl-devel, fontconfig-devel, freetype-devel, openssl-devel, glibc-devel, libicu-devel
+Requires:       libcurl, fontconfig, freetype, openssl, glibc libicu
+# Requirements not packaged in main repos
+# COPR @dotnet-sig/dotnet
+BuildRequires:  dotnet-sdk-2.2
+# RPMfusion free
+Requires:       ffmpeg
+
+# For the update-db-paths.sh script to fix emby paths to jellyfin
+%{?fedora:Recommends: sqlite}
+
+# Fedora has openssl1.1 which is incompatible with dotnet 
+%{?fedora:Requires: compat-openssl10}
+# Disable Automatic Dependency Processing for Centos
+%{?el7:AutoReqProv: no}
+
+%description
+Jellyfin is a free software media system that puts you in control of managing and streaming your media.
+
+
+%prep
+%autosetup -n %{name}-%{version}
+pushd ThirdParty
+    tar xf %{S:5}
+    rm -rf taglib-sharp
+    mv taglib-sharp-%{taglib_commit} taglib-sharp
+popd
+
+%build
+export DOTNET_CLI_TELEMETRY_OPTOUT=1
+export DOTNET_SKIP_FIRST_TIME_EXPERIENCE=1
+dotnet build --runtime linux-x64
+
+%install
+export DOTNET_CLI_TELEMETRY_OPTOUT=1
+export DOTNET_SKIP_FIRST_TIME_EXPERIENCE=1
+dotnet publish --configuration Release --output='%{buildroot}%{_libdir}/jellyfin' --self-contained --runtime linux-x64
+%{__install} -D -m 0644 LICENSE %{buildroot}%{_datadir}/licenses/%{name}/LICENSE
+%{__install} -D -m 0644 debian/conf/jellyfin.service.conf %{buildroot}%{_sysconfdir}/systemd/system/%{name}.service.d/override.conf
+%{__install} -D -m 0644 debian/conf/logging.json %{buildroot}%{_sysconfdir}/%{name}/logging.json
+%{__mkdir} -p %{buildroot}%{_bindir}
+tee %{buildroot}%{_bindir}/jellyfin << EOF
+#!/bin/sh
+exec %{_libdir}/%{name}/%{name} \${@}
+EOF
+%{__mkdir} -p %{buildroot}%{_sharedstatedir}/jellyfin
+%{__mkdir} -p %{buildroot}%{_sysconfdir}/%{name}
+%{__mkdir} -p %{buildroot}%{_var}/log/jellyfin
+
+%{__install} -D -m 0644 %{SOURCE1} %{buildroot}%{_unitdir}/%{name}.service
+%{__install} -D -m 0644 %{SOURCE2} %{buildroot}%{_sysconfdir}/sysconfig/%{name}
+%{__install} -D -m 0600 %{SOURCE3} %{buildroot}%{_sysconfdir}/sudoers.d/%{name}-sudoers
+%{__install} -D -m 0750 %{SOURCE4} %{buildroot}%{_libexecdir}/%{name}/restart.sh
+%{__install} -D -m 0755 %{SOURCE6} %{buildroot}%{_datadir}/%{name}/update-db-paths.sh
+%{__install} -D -m 0755 %{SOURCE7} %{buildroot}%{_prefix}/lib/firewalld/service/%{name}.xml
+
+%files
+%{_libdir}/%{name}/dashboard-ui/*
+%attr(755,root,root) %{_bindir}/%{name}
+%attr(644,root,root) %{_libdir}/%{name}/*.json
+%attr(644,root,root) %{_libdir}/%{name}/*.pdb
+%attr(755,root,root) %{_libdir}/%{name}/*.dll
```

Comment:
is there a reason for setting `executable` on `.dll`, `.so` and `.a` files?

Reply:
@JustAMan Now i re-encountered the problem i had originally and the reason why i set all the binaries to `755`.

For some reason has the main `jellyfin` binary `0766` permissions after the build with `dotnet` (nothing special just `dotnet publish`), so when jellyfin runs as service which is run under the `jellyfin` user it gets some permission denied errors.